elf-loader.core

```diff
@@ -1,23 +1,44 @@
-CAPI=1
-[main]
-name = ::elf-loader:1.0.2
-description = Generic ELF loader
-
-[vpi]
-src_files = elf-loader.c vpi_wrapper.c
-include_files = elf-loader.h
-libs = -lelf
-
-[verilator]
-src_files     = elf-loader.c
-include_files = elf-loader.h
-libs          = -lelf
-
-[parameter elf-load]
-datatype    = file
-description = ELF file to preload to memory
-paramtype   = plusarg
-scope       = public
-
-[scripts]
-pre_build_scripts = check_libelf.sh
+CAPI=2:
+
+name : ::elf-loader:1.0.3
+description : Generic ELF loader
+
+filesets :
+  elf_loader :
+    files : [elf-loader.c, elf-loader.h : {is_include_file : true}]
+    file_type : cSource
+
+  vpi :
+    files : [vpi_wrapper.c]
+    file_type : cSource
+
+  check_libelf:
+    files: [check_libelf.sh : {file_type: user, copyto: check_libelf.sh}]
+
+vpi :
+  elf_loader_vpi :
+    filesets : [elf_loader, vpi]
+    libs : [elf]
+
+targets :
+  default :
+    hooks :
+      pre_build : [check_libelf]
+    filesets : ["tool_verilator? (elf_loader)", check_libelf]
+    parameters : ["tool_icarus? (elf_load)"]  # for verilator the argument is --elf-load, not a plusarg
```

Comment:
It sounds like this should be `"!tool_verilator? (elf_load)"` instead

Reply:
Yeah, that is better, not sure what i was thinking